includes/bootstrap.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * Bootstrap file for the AI plugin.
+ *
+ * Handles plugin initialization, version checks, and feature loading.
+ *
+ * @package WordPress\AI
+ */
+
+namespace WordPress\AI;
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+// Define plugin constants.
+if ( ! defined( 'AI_VERSION' ) ) {
+	define( 'AI_VERSION', '0.1.0' );
+}
+if ( ! defined( 'AI_PLUGIN_FILE' ) ) {
+	define( 'AI_PLUGIN_FILE', defined( 'WP_AI_DIR' ) ? WP_AI_DIR . 'ai.php' : '' );
+}
+if ( ! defined( 'AI_PLUGIN_DIR' ) ) {
+	define( 'AI_PLUGIN_DIR', defined( 'WP_AI_DIR' ) ? WP_AI_DIR : '' );
+}
+if ( ! defined( 'AI_PLUGIN_URL' ) ) {
+	define( 'AI_PLUGIN_URL', plugin_dir_url( AI_PLUGIN_FILE ) );
+}
+if ( ! defined( 'AI_MIN_PHP_VERSION' ) ) {
+	define( 'AI_MIN_PHP_VERSION', '7.4' );
+}
+if ( ! defined( 'AI_MIN_WP_VERSION' ) ) {
+	define( 'AI_MIN_WP_VERSION', '6.9' );
+}
+
+/**
+ * Displays an admin notice for version requirement failures.
+ *
+ * @since 0.1.0
+ *
+ * @param string $message The error message to display.
+ */
+function version_notice( string $message ): void {
+	if ( ! is_admin() ) {
+		return;
+	}
+	?>
+	<div class="notice notice-error">
+		<p><?php echo esc_html( $message ); ?></p>
+	</div>
+	<?php
+}
+
+/**
+ * Checks if the PHP version meets the minimum requirement.
+ *
+ * @since 0.1.0
+ *
+ * @return bool True if PHP version is sufficient, false otherwise.
+ */
+function check_php_version(): bool {
+	if ( version_compare( phpversion(), AI_MIN_PHP_VERSION, '<' ) ) {
```

Comment:
A small thing but when testing this PR against WP 6.9-beta1, using this constant set to `6.9` causes the plugin to show an admin notice and stops the bootstrap process.

> AI plugin requires WordPress version 6.9 or higher. You are running WordPress version 6.9-beta1-61040-src.

I had to edit the value to make the plugin work during beta testing. Of course it's only a temporary inconvenience, but it made me wonder if we should use a different way, like checking if the `wp_register_ability()` function exists. Thoughts?